test/helpers/dashboard_helper_test.rb

```diff
@@ -89,65 +41,7 @@ class DashboardHelperTest < ActionView::TestCase
     assert_equal "This is a 100.0 % increase compared to the previous 10 days", tooltip_content(2, 1, 'Custom Range', (Time.now - 10.days).to_date, Time.now.to_date)
   end
 
-  test 'total watch time' do
-    mock_video_view_events
-    assert_equal 30000, total_watch_time(Ahoy::Event.where(id: [@video_watch_event_1, @video_watch_event_2].map(&:id)))
-  end
-
   test 'to_minutes' do
     assert_equal '10,000.0 min', to_minutes(10000*1000*60)
   end
-
-  test 'total_views' do
-    mock_video_view_events
-    assert_equal 2, total_views(Ahoy::Event.where(id: [@video_watch_event_1, @video_watch_event_2, @video_watch_event_3].map(&:id)))
-  end
-
-  test 'avg_view_duration' do
-    assert_equal 0, avg_view_duration([])
-    mock_video_view_events
-    assert_equal 25000.0, avg_view_duration(Ahoy::Event.where(id: [@video_watch_event_1, @video_watch_event_2, @video_watch_event_3].map(&:id)))
-  end
-
-  test 'avg_view_percentange' do
-    mock_video_view_events
-    assert_equal 50.0, avg_view_percentage(Ahoy::Event.where(id: [@video_watch_event_1, @video_watch_event_2, @video_watch_event_3].map(&:id)))
-  end
-
-  private 
-  
-  def mock_video_view_events
```

Comment:
@alis-khadka have these tests been replaced?

Reply:
Nope.

Some calculations like total_watch_time, total_views, avg_view_duration, avg_view_percentage were previously done with helper methods. Now, they have been moved to model methods since the calculations involve working on the table itself.

These tests were moved to model level tests with the following commit:
https://github.com/restarone/violet_rails/pull/1462/commits/2021e6a483ed1d46098eb22ee220b9035729020b